components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -46,6 +45,10 @@ class ContileTopSitesProvider(
     private val logger = Logger("ContileTopSitesProvider")
     private val diskCacheLock = Any()
 
+    // The last modified time of the disk cache.
+    @VisibleForTesting
+    internal var diskCacheLastModified: Long? = null
```

Comment:
We need to make this volatile as the writing and reading thread aren't the same, and we want to guarantee that readers see the latest value.